src/color.rs

```diff
@@ -420,6 +420,153 @@ impl Color {
         self.alpha = alpha;
         self
     }
+
+    /// Mix this color with the other given color, with the given weighting .
+    /// 
+    /// The weight determines the amount of the first color that will be used
+    /// in mixing as a percentage. So 50% means use each color equally, while 
+    /// 25% indicates to use one-quarter of this color, and seventy-five percent 
+    /// of the second.
+    /// 
+    /// The alpha channel of both colors are also combined in this way.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let c1 = Color::rgba(18., 55., 125., 1.0);
+    /// let mixed = c1.mix(&Color::rgba(125., 33., 200., 0.8), 40);
+    /// assert_eq!(mixed, Color::rgba(72., 44., 163., 0.88));
+    /// ```
+    /// 
+    /// Passing a value for `weight` that is not between 0 and 100 will result
+    /// in `Err(String)` being returned.
```

Comment:
This sentence seems to require an update. The method will panic if `weight` is not a value between 0 and 100. We should also consider panicking if `Color` is not valid.